src/core/Password.cpp

```diff
@@ -11,20 +13,70 @@
 #include <cstring>
 #include <thread>
 
-//
-int conv(int num_msg, const struct pam_message** msg, struct pam_response** resp, void* appdata_ptr) {
-    const char*          pass      = static_cast<const char*>(appdata_ptr);
-    struct pam_response* pam_reply = static_cast<struct pam_response*>(calloc(num_msg, sizeof(struct pam_response)));
-
-    for (int i = 0; i < num_msg; ++i) {
-        switch (msg[i]->msg_style) {
-            case PAM_PROMPT_ECHO_OFF:
-            case PAM_PROMPT_ECHO_ON: pam_reply[i].resp = strdup(pass); break;
-            case PAM_ERROR_MSG: Debug::log(ERR, "PAM: {}", msg[i]->msg); break;
-            case PAM_TEXT_INFO: Debug::log(LOG, "PAM: {}", msg[i]->msg); break;
+using namespace std;
+
+class AuthClient {
+  public:
+    int sock;
+    AuthClient(int sockfd) {
+        sock = sockfd;
+    };
+    //TODO: replace with std::expected errorhandling when clangd supports C++23
+    string question(const struct pam_message* msg) {
```

Comment:
can use `std::optional` I guess in the meantime

Reply:
ok, I will